samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py

```diff
@@ -0,0 +1,811 @@
+
+        By default this will just return the item if non of the types match. This is because of the function
+        resolve_all_attributes which will recreate the resources by processing every aspect of resource.
+
+        This code resolves in a top down depth first fashion in order to create a functional style recursion that
+        doesn't mutate any of the properties.
+
+        Parameters
+        ----------
+        intrinsic: dict, str, list, bool, int
+            This is an intrinsic property or an intermediate step
+        parent_function: str
+            In case there is a missing property, this is used to figure out where the property resolved is missing.
+        Return
+        ---------
+        The simplified version of the intrinsic function. This could be a list,str,dict depending on the format required
+        """
+        if intrinsic is None:
+            raise InvalidIntrinsicException("Missing Intrinsic property in {}".format(parent_function))
+
+        if any(isinstance(intrinsic, object_type) for object_type in
+               [string_types, list, bool, int]) or intrinsic == {}:
+            return intrinsic
+
+        keys = list(intrinsic.keys())
+        key = keys[0]
+
+        if key in self.intrinsic_key_function_map:
+            intrinsic_value = intrinsic.get(key)
+            return self.intrinsic_key_function_map.get(key)(intrinsic_value)
+        elif key in self.conditional_key_function_map:
+            intrinsic_value = intrinsic.get(key)
+            return self.conditional_key_function_map.get(key)(intrinsic_value)
+
+        # In this case, it is a dictionary that doesn't directly contain an intrinsic resolver, we must recursively
+        # resolve each of it's sub properties.
+        sanitized_dict = {}
+        for key, val in intrinsic.items():
+            sanitized_key = self.intrinsic_property_resolver(key, parent_function=parent_function)
+            sanitized_val = self.intrinsic_property_resolver(val, parent_function=parent_function)
+            verify_intrinsic_type_str(sanitized_key,
+                                      message="The keys of the dictionary {} in {} must all resolve to a string"
+                                      .format(sanitized_key, parent_function))
+            sanitized_dict[sanitized_key] = sanitized_val
+        return sanitized_dict
+
+    def resolve_template(self, template=None, symbol_resolver=None, ignore_errors=False):
+        """
+        This will parse through every entry in a CloudFormation template and resolve them based on the symbol_resolver.
+        Customers can optionally ignore resource errors and default to whatever the resource provides.
+
+        Parameters
+        -----------
+        template: dict
+            A dictionary containing all the resources and the attributes of the CloudFormation template. The class does
+            not mutate the template.
+        ignore_errors: bool
```

Comment:
Should `ignore_errors` ignore any and all errors/exceptions or only certain ones?

Reply:
I updated the docstring to reflect that ignore_errors only ignores InvalidIntrinsicExceptions